handle_linux.go

```diff
@@ -62,6 +66,15 @@ func newHandle(newNs, curNs netns.NsHandle, nlFamilies ...int) (*Handle, error)
 	return h, nil
 }
 
+// SetTimeout sets the amount of time the library will wait for the socket send
+// and for the socket receive on this handle to terminate before declaring a
+// a timeout failure to the client.
+func (h *Handle) SetTimeout(to time.Duration) {
```

Comment:
I wanted to avoid a precedence for adding a new constructor each time we add a new option.
This is more flexible and client will likely set the timeout once at handle creation. After that it is only going to be read locks.